### PR TITLE
Bug 1965283: Sync Static Resources with Cache

### PR DESCRIPTION
**Pull request by adambkaplan**
Refactor ocm-o to use the static resources controller to sync static
YAML assets. Added openshift-infra as a namespace to use an informer to
fetch core k8s objects.

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -206,6 +206,7 @@ github.com/openshift/library-go/pkg/operator/resource/resourcehelper
 github.com/openshift/library-go/pkg/operator/resource/resourcemerge
 github.com/openshift/library-go/pkg/operator/resource/resourceread
 github.com/openshift/library-go/pkg/operator/resourcesynccontroller
+github.com/openshift/library-go/pkg/operator/staticresourcecontroller
```
Comment (gabemontero) [inline vendor/modules.txt:209]: per discussion in office hours, @adambkaplan believes he can do some git refactoring to get this file's change minimally out of the main code change commit

File: pkg/operator/sync_openshiftcontrollermanager_v311_00.go [hunk 2]
```diff
@@ -38,41 +37,9 @@ func syncOpenShiftControllerManager_v311_00_to_latest(c OpenShiftControllerManag
...
-		if currResult.Error != nil {
-			errors = append(errors, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
-			continue
-		}
 
-		if currResult.Changed && resourcesThatForceRedeployment.Has(currResult.File) {
-			forceRollout = true
-		}
-	}
+	// TODO - why does a change to a ServiceAccount force redeployment of the daemonset?
```
Comment (gabemontero) [inline pkg/operator/sync_openshiftcontrollermanager_v311_00.go:41]: per discussion in office hours, this TODO can be removed

File: vendor/github.com/openshift/library-go/pkg/operator/staticresourcecontroller/static_resource_controller.go [hunk 3]
```diff
@@ -0,0 +1,240 @@
+package staticresourcecontroller
```
Comment (gabemontero) [inline vendor/github.com/openshift/library-go/pkg/operator/staticresourcecontroller/static_resource_controller.go:1]: per discussion in office hours, @adambkaplan believes he can do some git refactoring to get this file's change minimally out of the main code change commit

**Review by gabemontero — CHANGES_REQUESTED** [review]
"officially" recording for posterity @adambkaplan what we discussed in offices hours today

of course let me know if something unexpected arises and the commit organizational changes become untenable

**Comment by adambkaplan** [comment]
/close

Going to implement in #216 instead

**Pull request closed without merging**